src/notebookMiddlewareAddon.ts

```diff
@@ -132,6 +135,39 @@ export class NotebookMiddlewareAddon implements Middleware, Disposable {
         this.converter.dispose();
     }
 
+    public stopWatching(notebook: NotebookDocument): void {
+        // Just close the document. This should cause diags and other things to be cleared
+        const client = this.getClient();
+        if (client && notebook.cellCount > 0) {
+            const outgoing = this.converter.toOutgoingDocument(notebook.cellAt(0).document);
+            const params = client.code2ProtocolConverter.asCloseTextDocumentParams(outgoing);
+            client.sendNotification(DidCloseTextDocumentNotification.type, params);
+
+            // Set the diagnostics to nothing for all the cells
+            if (client.diagnostics) {
+                notebook.getCells().forEach(c => {
+                    client.diagnostics?.set(c.document.uri, []);
+                })
+            }
+
+            // Remove from tracking by the converter
+            this.converter.remove(notebook.cellAt(0).document);
```

Comment:
Is it ok using just the first cell document here?

Reply:
Actually just reading more looks like any cell's text document can be used to index into the converter. Disregard :)